### ISSUE TITLE
Instance Segmentation does not show terrain

### ISSUE THREAD

**Issue by yoakim82** [carla-simulator/carla#6905]
CARLA version: 0.9.14 (custom build with included custom objects)
At least Town01 and Town03 affected, probably all maps.

Platform/OS: Ubuntu 22.04 LTS

Problem you have experienced:

When creating snapshots of a scene with both depth, rgb and segmentation sensors, there is an unexpected behavior in the segmentation view of the scene. Note I'm referring to the instance segmentation sensor here.
We use the instance segmentation to capture tight bounding boxes for our objects of interest, but when an object is spawned behind a hill or piece of terrain, the segmentation view will show the object even though it is 100% occluded in the rgb view. This totally ruins the purpose of having semantic segmentation. 

![terrain_traffic_light_rgb](https://github.com/carla-simulator/carla/assets/6775811/04c2492d-0df2-4dbe-9b43-d0c44af1e634)
![terrain_traffic_light_inst_seg](https://github.com/carla-simulator/carla/assets/6775811/f156fd70-d16a-49f4-96a7-316862808e38)

What you expected to happen:
I expect the objects being occluded by the terrain to not show up in the segmentation view, which seem to be the case for other occluding objects (e.g. trees/vegetation).
To clarify, the bounding boxes shown in the rgb image are generated only from information in the instance segmentation image.

Steps to reproduce:
Setup a triplet of sensors and navigate to a place in the map like in our enclosed scene.

**Comment by nikky4D** [comment]
did you have this solved?

**Comment by yoakim82** [comment]
No, I did not. My guess is it's related to how the town is created and how the terrain is labeled semantically.

But I have no clue as to how to fix this. Anyone with deeper UE4 skills is welcome to comment on this.

**Comment by Yongzhaooo** [comment]
I met the same situation and it is still not fixed now : (
![image](https://github.com/carla-simulator/carla/assets/133014490/a5852675-ca8e-4736-9dc7-f6756e094898)

**Comment by AreopagX** [comment]
related: #5367

**Comment by tdegraaff** [comment]
@jorge-kabuto Can now be closed (https://github.com/carla-simulator/carla/pull/8528).
